src/load-svg-string.js

```diff
@@ -0,0 +1,334 @@
+        halfStrokeWidth = 0;
+    } else {
+        halfStrokeWidth = findLargestStrokeWidth(svgTag) / 2;
+    }
+    const width = bbox.width + (halfStrokeWidth * 2);
+    const height = bbox.height + (halfStrokeWidth * 2);
+    const x = bbox.x - halfStrokeWidth;
+    const y = bbox.y - halfStrokeWidth;
+
+    // Set the correct measurements on the SVG tag
+    svgTag.setAttribute('width', width);
+    svgTag.setAttribute('height', height);
+    svgTag.setAttribute('viewBox',
+        `${x} ${y} ${width} ${height}`);
+};
+
+/**
+ * Find all instances of a URL-referenced `stroke` in the svg. In 2.0, all gradient strokes
+ * have a round `stroke-linejoin` and `stroke-linecap`... for some reason.
+ * @param {SVGSVGElement} svgTag the SVG tag to apply the transformation to
+ */
+const setGradientStrokeRoundedness = svgTag => {
+    const elements = collectElements(svgTag);
+
+    for (const elt of elements) {
+        if (!elt.style) continue;
+        const stroke = elt.style.stroke || elt.getAttribute('stroke');
+        if (stroke && stroke.match(/^url\(#.*\)$/)) {
+            elt.style['stroke-linejoin'] = 'round';
+            elt.style['stroke-linecap'] = 'round';
+        }
+    }
+};
+
+/**
+ * In-place, convert passed SVG to something consistent that will be rendered the way we want them to be.
+ * Currently, this will normalize stroke widths (see transform-applier.js) and render all embedded images pixelated.
+ * The returned SVG will be guaranteed to always have a `width`, `height` and `viewBox`.
+ * In addition, if the `fromVersion2` parameter is `true`, several "quirks-mode" transformations will be applied which
+ * mimic Scratch 2.0's SVG rendering.
+ * @param {SVGSvgElement} svgTag root SVG node to operate upon
+ * @param {boolean} [fromVersion2] True if we should perform conversion from version 2 to version 3 svg.
+ */
+const normalizeSvg = (svgTag, fromVersion2) => {
+    if (fromVersion2) {
+        // Fix gradients. Scratch 2 exports no x2 when x2 = 0, but
+        // SVG default is that x2 is 1. This must be done before
+        // transformStrokeWidths since transformStrokeWidths affects
+        // gradients.
+        transformGradients(svgTag);
+    }
+    transformStrokeWidths(svgTag, window);
+    transformImages(svgTag);
+    if (fromVersion2) {
+        // Transform all text elements.
+        transformText(svgTag);
+        // Transform measurements.
+        transformMeasurements(svgTag);
+        // Fix stroke roundedness.
+        setGradientStrokeRoundedness(svgTag);
+    } else if (!svgTag.getAttribute('viewBox')) {
+        // Renderer expects a view box.
+        transformMeasurements(svgTag);
+    } else if (!svgTag.getAttribute('width') || !svgTag.getAttribute('height')) {
+        svgTag.setAttribute('width', svgTag.viewBox.baseVal.width);
+        svgTag.setAttribute('height', svgTag.viewBox.baseVal.height);
+    }
+};
+
+/**
+ * Load an SVG string and normalize it. All the steps before drawing/measuring.
```

Comment:
note to self: this should probably include the JSDoc blurb from above that documents what a "normalized" SVG actually guarantees, as this is the actual public exported function